web/src/pages/FilesPage.tsx

```diff
@@ -51,15 +105,12 @@ export const FilesPage: React.FC = () => {
   }, [libraries, libraryId, path]);
 
   useEffect(() => {
-    refresh();
+    refresh().catch((e) => {
+      console.error(e);
+    });
   }, [refresh]);
 
-  useEffect(() => {
-    const id = parseInt(libraryId);
-    setLibrary(librariesList?.find((l) => l.id === id));
-  }, [librariesList, libraryId]);
-
-  useEffect(() => setSelectedEntry(undefined), [path]);
+  useEffect(() => setSelectedEntry(undefined), [setSelectedEntry, path]);
```

Comment:
One thing I literally just noticed during testing (also on main, but I figured might as well add that here): when you have a entry selected and you switch to a different library, that entry will remain selected, which feels a bit weird. Simple fix though:

```suggestion
  useEffect(() => setSelectedEntry(undefined), [setSelectedEntry, path, libraryId]);
```